### ISSUE TITLE
Gwt/Errai, graph not displaying

### ISSUE THREAD

**Issue by ps23** [jacomyal/sigma.js#649]
I am trying to use sigma.js in a gwt/Errai app. We use the same code to run d3 and it works fine, but for some reason, whatever I try, the sigma graph is not shown. With the basic example I do not get an error, and when adding a webgl renderer manually I get an error that the container was not found. 

The basic sample I use in the following way (again, the d3.select(el) counterpart works fine):

	@PostConstruct
	void timeViewInit() {
		visWidget.getElement().addClassName("graphWrapper");
	}
	
	public void drawGraph()
	{
		drawGraph(visWidget.getElement());
	}
	
	// JSNI
	private native void drawGraph(Element el)
	/*-{
	 
    var s = new sigma(el);

    s.graph.addNode({
      // Main attributes:
      id: 'n0',
      label: 'Hello',
      // Display attributes:
      x: 0,
      y: 0,
      size: 1,
      color: '#f00'
    }).addNode({
      // Main attributes:
      id: 'n1',
      label: 'World !',
      // Display attributes:
      x: 1,
      y: 1,
      size: 1,
      color: '#00f'
    }).addEdge({
      id: 'e0',
      // Reference extremities:
      source: 'n0',
      target: 'n1'
    });

    // Finally, let's ask our sigma instance to refresh:
    s.refresh();

	}-*/;

}

The DOM looks as follows:

![dom](https://cloud.githubusercontent.com/assets/699423/9873221/87e3a536-5b9e-11e5-9a98-73845c151c57.png)

The CSS:

.visWidget {
    position: absolute;
    top: 20px;
    bottom: 20px;
    width: 100%;
}

**Comment by Yomguithereal** [comment]
Hello @ps23, I am not acquainted with gwt/errai so this is a bit hard for me to debug this. Is the `el` variable in your JS the correct DOM element and is your JS correctly run?

**Comment by ps23** [comment]
Yes, if I use d3, then it works. We are pretty sure now actually that the problem is the HTMLElement check. We currently try to work around that and add the sigma.min.js manually to the web site, will report back.

**Comment by Yomguithereal** [comment]
@ps23 does GWT uses an abstraction over the DOM and fiddle with the `HTMLElement` object specifically? If so, yes this is a problem as sigma expects raw DOM.

**Comment by ps23** [comment]
No, but the standard for gwt is to load js code into the application context inside an iframe, and I am not really sure why, although I assume that they have good reasons to do so. There is a choice though to load the js into root which we do now to make it work, but I understand that this is the less preferred option. I did not look into d3, but it might be good if sigma used the same way to access the container as d3 does it, however, not sure how much work that entails...

**Comment by Yomguithereal** [comment]
There are no significant differences between the way both d3 and sigma access the containers upon which they act. Is your code open source somewhere? I would need a bit more than the example above to debug this and help you.

**Comment by ps23** [comment]
The code is not open source unfortunately. We solved it though:

So the problem is that sigma is not able to access the container from within the document iframe that gwt adds js libraries into by standard. 

So one has to inject the js script in a way that the global variables it defines are accessible through $wnd, one has to setWindow(ScriptInjector.TOP_WINDOW) in the ScriptInjector. http://stackoverflow.xluat.com/questions/14446136/gwt-javascript-injection-and-jsni

This has the effect that one has to adapt the sigma samples to use $wnd instead.

No sure why, but with d3, the TOP_WINDOW option is not necessary.

**Comment by archimagogabo** [comment]
Please an example of com $ wnd, by setWindow (ScriptInjector.TOP_WINDOW)
